data_models/complex_encoder.py

```diff
@@ -0,0 +1,11 @@
+import json
+
+'''Returns a JSON representation of a complex object, 
+by nesting JSON of other objects appropriately'''
+#code citation: https://stackoverflow.com/questions/5160077/encoding-nested-python-object-in-json
+class ComplexEncoder(json.JSONEncoder):
```

Comment:
Update for p[ython docstring format](https://www.python.org/dev/peps/pep-0257/)
```suggestion
#code citation: https://stackoverflow.com/questions/5160077/encoding-nested-python-object-in-json
class ComplexEncoder(json.JSONEncoder):
    """Returns a JSON representation of a complex object
     
    Handles nested JSON of other objects appropriately"""
```